flux_local/git_repo.py

```diff
@@ -772,20 +772,26 @@ async def update_kustomization(cluster: Cluster) -> None:
 
 
 @contextlib.contextmanager
-def create_worktree(repo: git.repo.Repo) -> Generator[Path, None, None]:
+def create_worktree(repo: git.repo.Repo, branch: str | None = None) -> Generator[Path, None, None]:
```

Comment:
Can you add a comment in the pydoc that `branch` is an existing branch? (or variable could be named `existing_branch`). I just want to clarify here that it is not creating a new branch since much of the code below is about avoiding creating a new branch, implying that it may be creating a branch when passed in.

Reply:
I renamed the argument to `existing_branch` like you suggested and added a comment.